### PR TITLE
update CHANGES.md and MIGRATION.md

### PR DESCRIPTION
**Pull request by OwenKephart**
### Summary & Motivation

There's some overlap between the Breaking Changes / Deprecations section of CHANGES.md and MIGRATION.md. I think this is ok (good to surface stuff in multiple places).

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -13,9 +13,14 @@ When new releases include breaking changes or deprecations, this document descri
 ### Legacy API Removals
 - Dagster's legacy APIs, which were marked "legacy" in 0.13.0, have been removed. This includes `@solid`, `SolidDefinition`, `@pipeline`, `PipelineDefinition`, `@composite_solid`, `CompositeSolidDefinition`, `ModeDefinition`, `PresetDefinition`, `PartitionSetDefinition`, `InputDefinition`, `OutputDefinition`, `DynamicOutputDefinition`, `pipeline_failure_sensor`, `@hourly_schedule`, `@daily_schedule`, `@weekly_schedule`, and `@monthly_schedule`. [Here is a guide](https://docs.dagster.io/0.15.6/guides/dagster/graph_job_op) to migrating from the legacy APIs to the stable APIs.
 - Deprecated arguments to library ops have been switched to reflect stable APIs. This includes `input_defs`/`output_defs` arguments on `define_dagstermill_op`, which have been changed to `ins`/`outs` respectively, and `input_defs` argument on `create_shell_script_op`, which has been changed to `ins`.
+- The `pipeline_selection` argument has been removed from `run_failure_sensor` and related decorators / functions. Instead, use `job_selection`.
```
Comment (jamiedemaria) [inline MIGRATION.md:16]: job_selection will cause a deprecation warning too, the new-new arg is `monitored_jobs`

additionally, `pipeline_failure_sensor` has been removed

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,66 @@
...
+* Ops and software-defined assets with the `None` return type annotation are now given the `Nothing` type instead of the `Any` type.
+* Fixed a bug that caused `AssetsDefinition.from_graph` and `from_op` to fail when invoked on a `configured` op.
+* The `materialize` function, which is not experimental, no longer emits an experimental warning.
+* Fixed a bug where runs from different repositories would be intermingled when viewing the runs for a specific repository-scoped job/schedule/sensor.
+* [dagster-dbt] A regression was introduced in 0.15.8 that would cause dbt logs to show up in json format in the UI. This has been fixed.
+* [dagster-databricks] Previously, if you were using the `databricks_pyspark_step_launcher`, and the external step failed to start, a `RESOURCE_DOES_NOT_EXIST` error would be surfaced, without helpful context. Now, in most cases, the root error causing the step to fail will be surfaced instead.
+
+### Documentation
+
+* New [guide](https://docs.dagster.io/guides/dagster/transitioning-data-pipelines-from-development-to-production) that walks through seamlessly transitioning code from development to production environments
```
Comment (jamiedemaria) [inline CHANGES.md:61]:
```suggestion
* New [guide](https://docs.dagster.io/guides/dagster/transitioning-data-pipelines-from-development-to-production) that walks through seamlessly transitioning code from development to production environments.
```
I forgot the . in my changelog entry sry

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,66 @@
...
+* The `materialize` function, which is not experimental, no longer emits an experimental warning.
+* Fixed a bug where runs from different repositories would be intermingled when viewing the runs for a specific repository-scoped job/schedule/sensor.
+* [dagster-dbt] A regression was introduced in 0.15.8 that would cause dbt logs to show up in json format in the UI. This has been fixed.
+* [dagster-databricks] Previously, if you were using the `databricks_pyspark_step_launcher`, and the external step failed to start, a `RESOURCE_DOES_NOT_EXIST` error would be surfaced, without helpful context. Now, in most cases, the root error causing the step to fail will be surfaced instead.
+
+### Documentation
+
+* New [guide](https://docs.dagster.io/guides/dagster/transitioning-data-pipelines-from-development-to-production) that walks through seamlessly transitioning code from development to production environments 
+* New [guide](https://docs.dagster.io/guides/dagster/branch_deployments) that demonstrates using Branch Deployments to test Dagster code in your cloud environment without impacting your production data
```
Comment (jamiedemaria) [inline CHANGES.md:62]:
```suggestion
* New [guide](https://docs.dagster.io/guides/dagster/branch_deployments) that demonstrates using Branch Deployments to test Dagster code in your cloud environment without impacting your production data.
```

File: CHANGES.md [hunk 4]
```diff
@@ -1,5 +1,66 @@
...
+In 1.0.0, a large number of previously-deprecated APIs have been fully removed. A full list of breaking changes and deprecations, alongside instructions on how to migrate older code, can be found in [MIGRATION.md](https://github.com/dagster-io/dagster/blob/master/MIGRATION.md). At a high level:
+
+* The `solid` and `pipeline` APIs have been removed, along with references to them in extension libraries, arguments, and the CLI *(deprecated in `0.13.0)`*.
+* The `AssetGroup` and `build_asset_job` APIs, and a host of deprecated arguments to asset-related functions, have been removed *(deprecated in `0.15.0`)*.
+* The `EventMetadata` and `EventMetadataEntryData` APIs have been removed *(deprecated in `0.15.0`)*.
+
+### Deprecations
+
+* `dagster_type_materializer` and `DagsterTypeMaterializer` have been marked experimental and will likely be removed within a 1.x release. Instead, use an `IOManager`.
+* `FileManager` and `FileHandle` have been marked experimental and will likely be removed within a 1.x release. 1.x release.
```
Comment (sryza) [inline CHANGES.md:33]: 1.x release repeated twice

File: CHANGES.md [hunk 5]
```diff
@@ -1,5 +1,66 @@
...
+
+### Other Changes
+
+* As of 1.0.0, Dagster no longer guarantees support for python 3.6. This is in line with [PEP 494](https://peps.python.org/pep-0494/), which outlines that 3.6 has reached end of life.
+* **[planned]** In an upcoming 1.x release, we plan to make a change that renders values supplied to `configured` in Dagit. Up through this point, values provided to `configured` have not been sent anywhere outside the process where they were used. This change will mean that, like other places you can supply configuration, `configured` is not a good place to put secrets: **You should not include any values in configuration that you don't want to be stored in the Dagster database and displayed inside Dagit.**
+* `fs_io_manager`, `s3_pickle_io_manager`, and `gcs_pickle_io_manager`, and `adls_pickle_io_manager` no longer write out a file or object when handling an output with the `None` or `Nothing` type.
+* The `custom_path_fs_io_manager` has been removed, as its functionality is entirely subsumed by the `fs_io_manager`, where a custom path can be specified via config.
+* The default `typing_type` of a `DagsterType` is now `typing.Any` instead of `None`.
+
+### New since 0.15.8
```
Comment (sryza) [inline CHANGES.md:43]: This should be H2, right?

**Review by sryza — CHANGES_REQUESTED** [review]

File: CHANGES.md [hunk 6]
```diff
@@ -1,5 +1,66 @@
...
+* A new `dagster project `CLI contains commands for bootstrapping new Dagster projects and repositories
+    * `dagster project scaffold` creates a folder structure with a single Dagster repository and other files such as workspace.yaml. This CLI enables you to quickly start building a new Dagster project with everything set up.
+    * `dagster project from-example` downloads one of the Dagster examples. This CLI helps you to quickly bootstrap your project with an officially maintained example. You can find the available examples via `dagster project list-examples`.
+    * Check out [Create a New Project](https://docs.dagster.io/getting-started/create-new-project) for more details.
+* A `default_executor_def` argument has been added to the `@repository` decorator. If specified, this will be used for any jobs (asset or op) which do not explicitly set an `executor_def`.
+* A `default_logger_defs` argument has been added to the `@repository` decorator, which works in the same way as `default_executor_def`.
+* A new `execute_job` function presents a new Python API for kicking off runs of your jobs.
```
Comment (sryza) [inline CHANGES.md:15]: "new" twice is a little weird? not a big deal though

**Review by sryza — APPROVED** [review]

File: CHANGES.md (hunk 6 above)
Comment (OwenKephart) [reply CHANGES.md:15]: good catch

**Pull request merged**